### PR TITLE
added test cases for APIs

### PR DESCRIPTION
**Pull request by salabh-aot**
- Added test cases for the APIs
  - code API
  - meta API
  - ops API
  - outcome API
  - staff API
- updates
  - renamed `work_status` to `work_statuses`
  - updated staff and milestone API endpoints to avoid misleading
  - updated sync form data API to properly handle singular vs plural nouns
  - refactored test cases for the URL changes
  - added `sync-form-data` update test case

### CODE REVIEW

File: src/reports_api/resources/milestone.py [hunk 1]
```diff
@@ -24,12 +24,12 @@
 
 
 @cors_preflight('GET')
-@API.route('/<int:phase_id>', methods=['GET', 'OPTIONS'])
+@API.route('/phase/<int:phase_id>', methods=['GET', 'OPTIONS'])
```
Comment (dinesh-aot) [inline src/reports_api/resources/milestone.py:27]: /phases/<int:phase_id> would be more appropriate I think. Let me know what do you think.

File: src/reports_api/resources/staff.py [hunk 2]
```diff
@@ -25,16 +25,36 @@
...
+class Staff(Resource):
+    """Endpoint resource to return staff details."""
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def get(id):
+        """Return a staff detail based on id."""
+        return StaffService.find_by_id(id), HTTPStatus.OK
+
+
+@API.route('/position/<int:position_id>', methods=['GET', 'OPTIONS'])
```
Comment (dinesh-aot) [inline src/reports_api/resources/staff.py:53]: same here. positions.

**Review by dinesh-aot — APPROVED** [review]

File: src/reports_api/resources/staff.py (hunk 2 above)
Comment (salabh-aot) [reply src/reports_api/resources/staff.py]: Updated

File: src/reports_api/resources/milestone.py (hunk 1 above)
Comment (salabh-aot) [reply src/reports_api/resources/milestone.py]: updated

**Review by dinesh-aot — APPROVED** [review]
Looks good.

**Pull request merged**